flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -692,6 +693,7 @@ private void shutdownAsyncThreads() throws Exception {
 	 */
 	private void disposeAllOperators(boolean logOnlyErrors) throws Exception {
 		if (operatorChain != null && !disposedOperators) {
+			Exception disposalException = null;
 			for (StreamOperatorWrapper<?, ?> operatorWrapper : operatorChain.getAllOperators(true)) {
 				StreamOperator<?> operator = operatorWrapper.getStreamOperator();
 				if (!logOnlyErrors) {
```

Comment:
I think this branch could also gain from your change (currently, it disposes operators until 1st failure).

Reply:
> I think this `!logOnlyErrors` branch could also gain from your change.
> Currently, it disposes operators until 1st failure.

Yes, I agree. However, I was a little afraid to make the change from the safety perspective. It changes the behavior of the code, like what exception to throw, and the exit procedure.

For example, in some cases, the system is preferring a faster crash as soon as the first disposal occurs (it is just an example, not saying it would be like that).

Reply:
I don't think that failing fast this was intentional here. Anyways, you are already changing this behavior for `logOnlyErrors == false` case.

Reply:
cc @pnowojski 

whether to merge the `disposeAllOperators(true/false)` logic

Reply:
merge the logic as discussed offline.